src/Compilers/Core/MSBuildTask/Vbc.cs

```diff
@@ -1094,10 +1096,10 @@ protected override bool CallHostObjectToExecute()
         {
             Debug.Assert(this.HostObject != null, "We should not be here if the host object has not been set.");
 
-            IVbcHostObject vbcHostObject = this.HostObject as IVbcHostObject;
+            IVbcHostObject vbcHostObject = (IVbcHostObject)this.HostObject;
```

Comment:
>(IVbcHostObject)this.HostObject [](start = 43, length = 31)

Is there a reason to use explicit cast rather than `as`? If so, the asserts below no longer apply.